include/nil/actor/zk/commitments/batched_commitment.hpp

```diff
@@ -104,7 +104,26 @@ namespace nil {
                         return math::lagrange_interpolation(U_interpolation_points);
                     }
 
-                    std::vector<std::vector<typename field_type::value_type>> get_unique_points_list() const{
+                    // We call them singles in recursive verifier
+                    std::vector<typename field_type::value_type> get_unique_points(){
+                        std::vector<typename field_type::value_type> result;
+//                        std::unordered_set<typename field_type::value_type> result_set;
+
+                        for( auto const &[k, point_batch]:_points ){
+                            for( auto const &point_set: point_batch ){
+                                for( auto const &point:point_set ){
+                                    if( std::find(result.begin(), result.end(), point) == result.end() ){
```

Comment:
Maybe let's use std::set for such type of insert? It's mark that we have unique_points.

Reply:
I suggested this change in crypto3-zk, it's not done in crypto3-zk yet, as you can see they created some commented out unordered_set<typename field_type::value_type> result_set;.

Will do the same thing here, once it's done in crypto3-zk.